rcl/src/rcl/time.c

```diff
@@ -77,130 +73,127 @@ rcl_get_ros_time(void * data, rcl_time_point_value_t * current_time)
 }
 
 bool
-rcl_time_source_valid(rcl_time_source_t * time_source)
+rcl_clock_valid(rcl_clock_t * clock)
 {
-  if (time_source == NULL ||
-    time_source->type == RCL_TIME_SOURCE_UNINITIALIZED ||
-    time_source->get_now == NULL)
+  if (clock == NULL ||
+    clock->type == RCL_CLOCK_UNINITIALIZED ||
+    clock->get_now == NULL)
   {
     return false;
   }
   return true;
 }
 
 rcl_ret_t
-rcl_time_source_init(
-  enum rcl_time_source_type_t time_source_type, rcl_time_source_t * time_source
+rcl_clock_init(
+  enum rcl_clock_type_t clock_type, rcl_clock_t * clock
 )
 {
-  switch (time_source_type) {
-    case RCL_TIME_SOURCE_UNINITIALIZED:
+  switch (clock_type) {
+    case RCL_CLOCK_UNINITIALIZED:
       RCL_CHECK_ARGUMENT_FOR_NULL(
-        time_source, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-      rcl_init_generic_time_source(time_source);
+        clock, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
+      rcl_init_generic_clock(clock);
       return RCL_RET_OK;
     case RCL_ROS_TIME:
-      return rcl_ros_time_source_init(time_source);
+      return rcl_ros_clock_init(clock);
     case RCL_SYSTEM_TIME:
-      return rcl_system_time_source_init(time_source);
+      return rcl_system_clock_init(clock);
     case RCL_STEADY_TIME:
-      return rcl_steady_time_source_init(time_source);
+      return rcl_steady_clock_init(clock);
     default:
       return RCL_RET_INVALID_ARGUMENT;
   }
 }
 
 rcl_ret_t
-rcl_time_source_fini(rcl_time_source_t * time_source)
+rcl_clock_fini(rcl_clock_t * clock)
 {
-  switch (time_source->type) {
+  switch (clock->type) {
     case RCL_ROS_TIME:
-      return rcl_ros_time_source_fini(time_source);
+      return rcl_ros_clock_fini(clock);
     case RCL_SYSTEM_TIME:
-      return rcl_system_time_source_fini(time_source);
+      return rcl_system_clock_fini(clock);
     case RCL_STEADY_TIME:
-      return rcl_steady_time_source_fini(time_source);
-    case RCL_TIME_SOURCE_UNINITIALIZED:
+      return rcl_steady_clock_fini(clock);
+    case RCL_CLOCK_UNINITIALIZED:
     // fall through
     default:
       return RCL_RET_INVALID_ARGUMENT;
   }
 }
 
 rcl_ret_t
-rcl_ros_time_source_init(rcl_time_source_t * time_source)
+rcl_ros_clock_init(rcl_clock_t * clock)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(time_source, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-  rcl_init_generic_time_source(time_source);
-  time_source->data = calloc(1, sizeof(rcl_ros_time_source_storage_t));
-  time_source->get_now = rcl_get_ros_time;
-  time_source->type = RCL_ROS_TIME;
+  RCL_CHECK_ARGUMENT_FOR_NULL(clock, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
+  rcl_init_generic_clock(clock);
+  clock->data = calloc(1, sizeof(rcl_ros_clock_storage_t));
```

Comment:
while refactoring this, this function could also take an allocator to avoid native (c)alloc calls.